### PR TITLE
Fixed: Geocode once for split requests

### PR DESCRIPTION
**Pull request by bcosynot**
Requests with multiple tasks are being geocoded once per task. Fixing it so geocoding is done once per request, instead.

### CODE REVIEW

File: src/__mocks__/geo.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
```
Comment (bcosynot) [inline src/__mocks__/geo.js:1]: This is a way to mock user modules in jest - https://jestjs.io/docs/en/manual-mocks#mocking-user-modules
Required for mocking this in tests of `RequestService`. See line 3 of test/service/request-service.test.js

**Comment by bcosynot** [comment]
I don't have a lot of free time this week unfortunately, so not adding any new tests to accompany this PR. Most of the code involves calling other APIs so tests wouldn't be too interesting either way.

File: src/service/request-service.js [hunk 2]
```diff
@@ -2,14 +2,79 @@ const preconditions = require("preconditions").singleton();
...
+   * Resolve and update coordinates for requester's address
+   * @param request {RequestRecord} Request requiring coordinates
+   * @returns {Promise<RequestRecord>} request records updated with coordinates
+   * @throws error when unable to resolve coordinates or update them in airtable
+   */
+  async resolveAndUpdateCoords(request) {
+    preconditions.shouldBeObject(request);
+    preconditions.shouldBeString(request.fullAddress);
+    try {
+      if (request.coordinates) {
```
Comment (azlyth) [inline src/service/request-service.js:29]: A request address could be updated, so there has to be similar logic to the other coordinate retrieval (which saves the address for which coordinates were calculated from).

Comment (bcosynot) [reply src/service/request-service.js]: Just pushed a change addressing this change request.

**Comment by bcosynot** [comment]
@azlyth These changes require a couple of new columns in the prod "Requests" table.

1. `_coordinates` 
image

2. `_coordinates_address`
image

**Comment by azlyth** [comment]
Hm I see the line that does `_coordinates_address` update in the code but not seeing it happen when I ran through a case?

image

**Comment by bcosynot** [comment]
@azlyth That can happen if `_coordinates` was already populated. I can't think of another scenario to reproduce this.

**Comment by azlyth** [comment]
Both coordinate fields were blank though, hmm.

**Pull request merged**